### DISCUSSION TITLE
Analysis in .NET Core Projects Produces Different Results

### DISCUSSION THREAD

**Discussion by vsfeedback** [dotnet/roslyn#47913] [General]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1088041/analysis-in-net-core-projects-produces-different-r.html)._

---
After setting up code analysis in .NET Core, it's supposed to run analysis on every build. It does not seem to since the number of issues is a lot smaller than if I ran Code Analysis on Solution.

Also, sometimes if I click on a violation and go to the code, all of a sudden many violations just dissapear (without me changing anything).

---
#### Feedback Bot on 6/23/2020, 00:11 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Feedback Bot on 6/23/2020, 09:58 AM: 

<p>Thank you for sharing your feedback! Our teams prioritize action on product issues with broad customer impact. See details at: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq">https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq</a>. In case you need answers to common questions or need assisted support, be sure to use <a target='_blank' href="https://visualstudio.microsoft.com/vs/support/">https://visualstudio.microsoft.com/vs/support/</a>. We’ll keep you posted on any updates to this feedback.</p>

#### Feedback Bot on 7/31/2020, 00:19 AM: 

I have detected that for the last 35 days, this issue didn't have much product team activity and a very small amount of new votes or comments. Based on this, its severity, and affected area, it’s my experience that this issue is very unlikely to be fixed.
#### David McCarter on 8/3/2020, 08:08 AM: 

<p>My passion is code quality and not fixing this makes no sense. I will try to get more votes!</p>

#### Manish Vasani [MSFT] on 8/3/2020, 08:28 AM: 

<p>@<a target='_blank' href="https://developercommunity.visualstudio.com/users/167861/DavidMcCarter.html">David McCarter</a> would you be able to provide a more concrete repro, with expected and actual set of diagnostics? Also note that VS IDE comes with some in-build Code Style analyzers (IDExxxx diagnostics), which are by default only executed in VS IDE and not on command line build. You would need to manually install the CodeStyle NuGet package and configure the severity of IDExxxx diagnostics to warning/error for them to be reported at build time: <a target='_blank' href="https://dotnet.myget.org/feed/roslyn/package/nuget/Microsoft.CodeAnalysis.CSharp.CodeStyle">https://dotnet.myget.org/feed/roslyn/package/nuget/Microsoft.CodeAnalysis.CSharp.CodeStyle</a> (currently a pre-release package)</p>

#### Feedback Bot on 8/12/2020, 01:02 AM: 

We will close this report in 14 days because we don’t have enough information to investigate further. To keep the problem open, please provide the requested details.
#### David McCarter on 8/12/2020, 08:30 AM: 

<p>Yes, you can look at my open-source assembly here: </p><p><a target='_blank' href="https://github.com/RealDotNetDave/dotNetTips.Utility.Core/releases/tag/2020.8.9" target="_blank">https://github.com/RealDotNetDave/dotNetTips.Utility.Core/releases/tag/2020.8.9</a></p><p>I have also reproduced the same issue in other projects.</p><p>David</p>

#### Feedback Bot on 9/21/2020, 00:13 PM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

#### Manish Vasani on 9/21/2020, 02:46 PM: 

<p>Thanks for the repro. I am going to port it to github and add more context on the issue.</p>

---
### Original Solutions
(no solutions)

**Comment by mavasani** [comment]
The core issue here is that the analyzers are disabled for live analysis:

```xml
    <RunAnalyzersDuringLiveAnalysis>false</RunAnalyzersDuringLiveAnalysis>
    <RunAnalyzersDuringBuild>true</RunAnalyzersDuringBuild>
```

If user just executes one of the "Build" command or "Run Code Analysis" command, we end up with equal number of warnings from both. However, we don't seem to handle diagnostic de-duping between diagnostics from build and run code analysis commands, so "Build + Intellisense" shows duplicates. When any diagnostic is clicked to open a file, de-dupe gets performed and one of the duplicates is removed. Unfortunately, diagnostics coming from "Run Code Analysis" when `RunAnalyzersDuringLiveAnalysis` is set to false are not showing squiggles, so the user feels like diagnostics are missing, but they are reported in the error list. So, we need 2 fixes here:

1. De-dupe diagnostics from "Build" command and "Run Code Analysis" command
2. Show squiggles for diagnostics from "Run Code Analysis" command when `RunAnalyzersDuringLiveAnalysis` is set to false

**Reply by RealDotNetDave** [reply]
That seems to work. Thanks!
